_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+
+{% img blog/supabase-vue/supabase-create-project.png alt:"Create a project in Supabase" width:"600" %}{: .center-image }
+
+Take note of your `Project API Keys` and `Project Configuration` in the `Connecting to your new project` section of the project page. Specifically, you will need the `anon` `public` API key and the `URL`.
+
+### Create your table in Supabase
+
+Now add a new table to your database. In Supabase there are two ways to do this. You can use the SQL editor or you can use the GUI Table editor. I opted to use the Table editor to see how powerful the UI was and how easy it was to use. Once you open the table editor you can click **Create a new table**. Give the table the name `Entries` and a description if you want.
+
+Note that the `id` and `created_at` columns are prepopulated for you so you can skip these. Add the following columns to the table.
+
+> description: text
+> isPublic: bool
+> date: timestamptz
+> title: text
+
+Make each column non-nullable.  By default, Supabase makes all columns nullable. Then click **Save**. After a moment you will see your table in the _Table editor_ section.
+
+{% img blog/supabase-vue/supabase-create-table.png alt:"Create a table in supabase" width:"600" %}{: .center-image }
+
+### Populate some sample data
+
+Click on the table you just created in the table editor and click **Insert Row**. This will bring up an editor where you can add new data. I added 3 entries: two public and one private entry. Give your entries dates that are in your current month. For example, I'm writing this article in September 2022 so I used three dates in that month.
+
+{% img blog/supabase-vue/supabase-add-row.png alt:"Add data to supabase" width:"600" %}{: .center-image }
+
+After each row, click save and you should see the data you complete populated in the Table editor.
+
+## Create your Vue Application
+
+It's now time to create your Vue application. Start by running the following command.
+
+```sh
+npm init vue@3.3.2 OktaSupabase
+```
+
+This will create a new Vue application called `OktaSupabase` in a subfolder on your working directory with the same name. There will be several options in this command such as the project name, TypeScript support, JSX support, and others. See the image below for the options I've chosen. If you want to use some of these options feel free to turn them on.
+
+{% img blog/supabase-vue/vue-init-options.png alt:"Create your vue application" width:"600" %}{: .center-image }
+
+### Install your dependencies
+
+Now you can install the dependencies you need for this project.
+
+> > Tanay : trying something a little different here. If its no good let me know.
```

Comment:
- We should batch the commands into a single code block so that the readers have to copy/paste just once.
- Instead of different sub-headings, we could have a bulleted list since the explanations are one-liners and the sub-headings don't add any SEO value.

Reply:
[96fe00f422d64fbd155e93452326a29ba63ca382](https://github.com/nickolasfisher/okta-blog/commit/96fe00f422d64fbd155e93452326a29ba63ca382)